### ISSUE TITLE
Allow adding private comments to lines of code in the diff-view

### ISSUE THREAD

**Issue by Soleone** [Yatser/prettypullrequests#27]
First of all, I love this extension, it's a life saver for me for reviewing large pull requests and not lose my focus.

One last feature i'd like to see is to be able to add "in progress" comments. The use case is that for very large pull requests i sometimes have an idea that something needs to be changed but i can't articulate it fully yet. (e.g. "eww this is gonna mess up the code base we need to find a way to make this scale better"). only once i've looked through the full PR or even thought more about it i am able to write a public comment that is productive instead of just negative.

Optimally the way it works is something like this:

- [ ] i can click next to any line and a new comment icon pops up
- [ ] clicking opens textbox (i don't think markdown or preview is required, just nice to have)
- [ ] submitting stores the comment in local datastore (browser specific), this seems like the easiest solution (cross browser support with server side storage would obviously be nice, but seems like total overkill)
- [ ] comment display inline similar to official public github comments, but formatted slightly differently to signal it's internal only for now
- [ ] next time i open the diff the comments get displayed automatically like you'd expect

curious if you think that's a possibility. keep up the great work!

**Comment by Yatser** [comment]
Hey there,

So, I think it would be a possibility for sure, but not likely to be something I would be willing to tackle. I don't think it's a massive feature to develop, but I don't think it's trivial either.

That said though, if you would be willing to submit a PR for it, I would definitely look into merging it and releasing it for everyone. Or, you could also fork my repo and run your own version with this feature, or write another plugin to manage this feature.

By the way though, the comment on local storage - remember that this is a chrome extension, it relies entirely on the chrome browser, so there would be no need to consider cross browser support. There is a chrome extension specific API for interacting with local storage that the plugin already uses, so that would likely be the way to achieve the goal.

Thanks for supporting the extension!

**Comment by Soleone** [comment]
thanks for the reply! my time is currently pretty limited, but maybe i'll get to it eventually, then i'll create a PR and reference this issue.

**Comment by gholtslander-va** [comment]
I think this is now an official feature of pull requests? You can "start a review" which will mark your comments as pending until you decide to unleash them on your unsuspecting PR author. You can even close the page and come back and your pending comments will be waiting for you.

We can probably close this issue now.

**Comment by Soleone** [comment]
yep, agreed. that github feature release was like christmas

**Comment by gholtslander-va** [comment]
Sooooo good.
